src/app/features/posts/posts.component.html

```diff
@@ -0,0 +1,57 @@
+<app-loading *ngIf="loading"></app-loading>
+
+<div class="container py-3">
+  <div class="search d-flex justify-content-start gap-5 align-items-center">
+    <div class="child d-flex">
+      <img alt="envelope" src="assets/images/envelop.svg" width="30" class="pointer">
+      <h4 class="title mx-2">Posts Feed</h4>
+    </div>
+
+    <div class="child d-flex ml-5">
+      <div class="d-flex justify-content-between gap-2 align-items-center pointer">
+        <p class="title mb-0" (click)="seeAsTable = true" *ngIf="!seeAsTable">
+          Table View
+        </p>
+        <img src="assets/images/tabla-active.svg" alt="table" class="mx-2" (click)="seeAsTable = true"
+          *ngIf="!seeAsTable">
+      </div>
+
+      <div class="d-flex justify-content-between gap-2 align-items-center pointer">
+        <p class="title mb-0" *ngIf="seeAsTable" (click)="seeAsTable = false">
+          Cards View
+        </p>
+        <img src="assets/images/grid-active.svg" alt="cards" (click)="seeAsTable = false" *ngIf="seeAsTable">
+      </div>
+    </div>
+  </div>
+
+  <div class="mt-3">
+    @if (postsResponse) {
+    @if (seeAsTable) {
+    <app-post-table [posts]="postsResponse" (viewDetails)="selectedPost = $event">
+    </app-post-table>
+    }
+    @else {
+    <div class="parent mt-3">
+      @for (post of postsResponse; track $index) {
+      <div class="child">
+        <app-post-card [post]="post" (click)="selectPost(post)">
+        </app-post-card>
+      </div>
+      }
+    </div>
+    }
+    }
+  </div>
+
+  <div class="my-5">
+    <app-pagination [total]="pagination.total" [pageSize]="pagination.pageSize"
+      (paginationChange)="onPaginationChange($event)">
+    </app-pagination>
+  </div>
+</div>
+
+@if(selectedPost) {
+<app-post-detail [post]="selectedPost" (close)="selectedPost = undefined">
+</app-post-detail>
+}
```

Comment:
Next time a dialog could be awesome. You can create a dialog factory and inject it to the template as well.

Reply:
> Next time a dialog could be awesome. You can create a dialog factory and inject it to the template as well.

Alright, I'll take into account this